dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java

```diff
@@ -1271,7 +1271,7 @@ private String getQueryOrderBy(
   private String getFromSubQueryLimitAndOffset(TrackedEntityQueryParams params) {
     StringBuilder limitOffset = new StringBuilder();
     int limit = params.getMaxTeLimit();
-    int teQueryLimit = systemSettingManager.getIntSetting(SettingKey.TRACKED_ENTITY_MAX_LIMIT);
+    int teQueryLimit = systemSettingManager.getIntSetting(SettingKey.TE_MAX_LIMIT);
```

Comment:
You are deprecating `TRACKED_ENTITY_MAX_LIMIT` but then you are just not using it anymore in the code.
What you should do is :
- if both keys are set, throw an error
- check which one is set and use that one

Reply:
-  if both are present // Error
-  if both are absent  // we use default configured for TE_MAX_LIMIT ?
-  if one is present // Use the one present